src/hotspot/share/gc/z/zMark.inline.hpp

```diff
@@ -28,14 +28,17 @@
 #include "gc/z/zMark.hpp"
 #include "gc/z/zMarkStack.inline.hpp"
 #include "gc/z/zThreadLocalData.hpp"
+#include "gc/z/zThread.inline.hpp"
 #include "runtime/thread.hpp"
 #include "utilities/debug.hpp"
 
 template <bool follow, bool finalizable, bool publish>
 inline void ZMark::mark_object(uintptr_t addr) {
   assert(ZAddress::is_marked(addr), "Should be marked");
   ZMarkThreadLocalStacks* const stacks = ZThreadLocalData::stacks(Thread::current());
-  ZMarkStripe* const stripe = _stripes.stripe_for_addr(addr);
+  ZMarkStripe* const stripe = push_local_stripe() && ZThread::is_worker() ?
```

Comment:
This introduces a volatile read in every object mark, do you have concurrent mark time mesaurement with this check?

Reply:
The volatile read has impact on the mark time, the average mark time increased by 4.7% 

Jdk17 zgc origin:
RUN RESULT: hbIR (max attempted) = 124943, hbIR (settled) = 121646, max-jOPS = 111199, critical-jOPS = 91446
RUN RESULT: hbIR (max attempted) = 122581, hbIR (settled) = 118652, max-jOPS = 111549, critical-jOPS = 92803
RUN RESULT: hbIR (max attempted) = 130653, hbIR (settled) = 119162, max-jOPS = 111055, critical-jOPS = 95202
[2021-03-26T16:49:41.474+0800][info ][gc,stats       ]       Phase: Concurrent Mark                               0.000 / 0.000      1142.885 / 1689.873    800.652 / 3473.744    800.652 / 3473.744    ms
[2021-03-26T19:03:55.173+0800][info ][gc,stats       ]       Phase: Concurrent Mark                               0.000 / 0.000      1204.003 / 1975.038    843.825 / 3006.138    843.825 / 3006.138    ms
[2021-03-26T21:14:13.816+0800][info ][gc,stats       ]       Phase: Concurrent Mark                               0.000 / 0.000      1147.566 / 2477.271    827.487 / 2982.508    827.487 / 2982.508    ms


Jdk17 zgc patch:
RUN RESULT: hbIR (max attempted) = 130653, hbIR (settled) = 116479, max-jOPS = 111055, critical-jOPS = 94165
RUN RESULT: hbIR (max attempted) = 130653, hbIR (settled) = 116752, max-jOPS = 109749, critical-jOPS = 97445
RUN RESULT: hbIR (max attempted) = 130653, hbIR (settled) = 118685, max-jOPS = 111055, critical-jOPS = 93540
[2021-03-30T20:53:35.593+0800][info ][gc,stats       ]       Phase: Concurrent Mark                               0.000 / 0.000      1198.332 / 3098.680    860.451 / 3198.799    860.451 / 3198.799    ms
[2021-03-30T23:07:36.532+0800][info ][gc,stats       ]       Phase: Concurrent Mark                               0.000 / 0.000      1155.819 / 1550.257    842.028 / 3189.933    842.028 / 3189.933    ms
[2021-03-31T01:27:26.091+0800][info ][gc,stats       ]       Phase: Concurrent Mark                               0.000 / 0.000      1211.280 / 1964.112    883.528 / 3053.926    883.528 / 3053.926    ms